extensions/2.0/Khronos/KHR_materials_clearcoat/README.md

```diff
@@ -61,29 +61,58 @@ All implementations should use the same calculations for the BRDF inputs. Implem
 |**clearcoatRoughnessFactor**      | `number`                                                                        | The clearcoat layer roughness.         | No, default: `0.0`   |
 |**clearcoatRoughnessTexture**     | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo)             | The clearcoat layer roughness texture. | No                   |
 |**clearcoatNormalTexture**        | [`normalTextureInfo`](/specification/2.0/README.md#reference-normaltextureinfo) | The clearcoat normal map texture.      | No                   |
-  
+
+
+The clearcoat effect is modeled via a microfacet-based BRDF. The BRDF is layered on top of the glTF 2.0 Metallic-Roughness material, including emission and all extensions.
+
+```
+# from glTF 2.0 Metallic-Roughness material (core)
+material = mix(dielectric_brdf, metal_brdf, metallic) + emission
+
+# coating BRDF
+clearcoat_brdf = specular_brdf((clearcoatRoughnessFactor * clearcoatRoughnessTexture.g)^2)
+
+# layer coating on top of base material
+coated_material = fresnel_coat(
+  layer = clearcoat_brdf,
+  base = material,
+  ior = 1.5,
+  weight = clearcoatTexture.r * clearcoatFactor,
+  normal = clearcoatNormalTexture)
```

Comment:
I'm a bit confused as to why normal is called out as an input into the fresnel_coat, but is not mentioned in specular_brdf.

Reply:
Fixed, now the clearcoat normal is passed to specular_brdf. It's still not perfect, the function as defined in Appendix B doesn't have this parameter. I am a bit afraid of making the normal a parameter everywhere, might be bad for readability.